osu.Game.Rulesets.Taiko/UI/TaikoMascotAnimation.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Audio.Track;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Animations;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Graphics.Containers;
+using osu.Game.Skinning;
+
+namespace osu.Game.Rulesets.Taiko.UI
+{
+    public sealed class TaikoMascotAnimation : BeatSyncedContainer
+    {
+        private readonly TextureAnimation textureAnimation;
+
+        private int currentFrame;
+
+        public TaikoMascotAnimation(TaikoMascotAnimationState state)
+        {
+            InternalChild = textureAnimation = createTextureAnimation(state).With(animation =>
+            {
+                animation.Origin = animation.Anchor = Anchor.BottomLeft;
+                RelativeSizeAxes = Axes.Both;
+            });
+
+            RelativeSizeAxes = Axes.Both;
+            Origin = Anchor = Anchor.BottomLeft;
+            AlwaysPresent = true;
```

Comment:
just so I don't forget, this is leftover from debugging attempts and should be removable safely

Reply:
I was super wrong - this is necessary for now, as `AnimationClockComposite` will consume time spent when not present otherwise, making `PlaybackPosition` jump forward in time.